packages/terra-time-input/src/TimeUtil.js

```diff
@@ -1,11 +1,143 @@
+   */
+  static parseTwelveHourTime(hour, anteMeridiem, postMeridiem) {
+    const parsedHour = { hourString: hour };
+    let numericHour = parseInt(hour, 10);
+
+    if (numericHour >= 12) {
+      parsedHour.meridiem = postMeridiem;
+
+      if (numericHour > 12) {
+        numericHour -= 12;
+      }
+    } else {
+      parsedHour.meridiem = anteMeridiem;
+
+      if (numericHour === 0) {
+        numericHour = 12;
+      }
+    }
+
+    if (numericHour < 10) {
+      parsedHour.hourString = '0'.concat(numericHour.toString());
+    } else {
+      parsedHour.hourString = numericHour.toString();
+    }
+
+    return parsedHour;
+  }
+
+  /**
+   * Increments an hour to it's next value. Resets to 01 if the passed in hour is 12 and the variant is a 12 hour clock.
+   * @param {String} hour Hour to increment
+   * @param {String} timeVariant Variant of time the passed in hour refers to.
+   * @return {String} Returns a string representation of the value of an hour after it's incremented. If the
+   *   hour is 01 and the timeVariant is 12, it returns 01.
+   */
+  static incrementHour(hour, timeVariant) {
+    const maxValue = timeVariant === this.FORMAT_12_HOUR ? 12 : 23;
+
+    // Increment the value by 1 when arrow up is pressed.
+    if (hour) {
+      let numericMinute = Number(hour);
+
+      if (numericMinute < maxValue) {
+        numericMinute += 1;
+        return numericMinute < 10 ? '0'.concat(numericMinute.toString()) : numericMinute.toString();
+      } else if (timeVariant === this.FORMAT_12_HOUR) {
+        return '01';
+      }
+
+      return hour;
+    }
+
+    return timeVariant === this.FORMAT_12_HOUR ? '12' : '00';
+  }
+
+  /**
+   * Decrements an hour to it's next value. Resets to 12 if the passed in hour is 01 and the variant is a 12 hour clock.
+   * @param {String} hour Hour to decrement
+   * @param {String} timeVariant Variant of time the passed in hour refers to.
+   * @return {String} Returns a string representation of the value of the an hour after it's decremented. If the
+   *   hour is 12 and the timeVariant is 12, it returns 12.
+   */
+  static decrementHour(hour, timeVariant) {
+    const minValue = timeVariant === this.FORMAT_12_HOUR ? 1 : 0;
+
+    if (hour) {
+      let numericMinute = Number(hour);
+
+      if (numericMinute > minValue) {
+        numericMinute -= 1;
+        return numericMinute < 10 ? '0'.concat(numericMinute.toString()) : numericMinute.toString();
+      } else if (timeVariant === this.FORMAT_12_HOUR) {
+        return '12';
+      }
+
+      return hour;
+    }
+
+    return timeVariant === this.FORMAT_12_HOUR ? '12' : '00';
+  }
+
+  /**
+   * Increments a minute to it's next value
+   * @param {String} minute Minute to increment
+   * @return {String} Returns a string representation of the value of the passed in minute after it's incremented
+   */
+  static incrementMinute(minute) {
+    if (minute) {
+      let numericMinute = Number(minute);
```

Comment:
what does this return if the value given is not numeric?

Reply:
Right now, the TimeInput doesn't recognize the value that are not in the format hh:mm has a whole, and resets it to blank when it's initialized with an invalid time value, so I think it's unlikely that this scenario will occur. https://github.com/cerner/terra-core/pull/1027/files/36678e4d3b233e232d19fd7d1565e17ff5e649e6#diff-662e6b853064e84accd761052bfc24c7R78

If you do want to have a check, I propose that if the minute is not a number after parsing, that we just reset the value to 00 or 12, as that's usually the default value for a clock.

Reply:
I'm happy with that it doesn't set the value as that is what I would expect us to do in this case.